### PR TITLE
Feature/makelife

### PR DESCRIPTION
**Pull request by cdcarter**
Updated the Makefile to run the tests the way we run them and release the package the way we release it. In the process, moved to the has-a-maintainer bump2version, and fixed up bumpversions configuration so that it doesn't create invalid python files.

At least wanted to get our current process documented in the repo and makefile before we change it.

# Critical Changes

# Changes

# Issues Closed

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -74,3 +74,24 @@ Before you submit a pull request, check that it meets these guidelines:
 * New classes, functions, etc have docstrings.
 * New code has comments.
 * Code style and file structure is similar to the rest of the project.
+* You have run the `black` code formatter.
+
+Releasing CumulusCI
+-------------------
+
+It's easy to release a version of CumulusCI to GitHub and PyPi! First, create a new branch for your version:
```
Comment (davisagli) [inline CONTRIBUTING.rst]: PyPI

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -74,3 +74,24 @@ Before you submit a pull request, check that it meets these guidelines:
...
+
+    $ git checkout -b feature/newversion
+
+After commiting any updates to HISTORY.rst and the docs, bump the version:
```
Comment (davisagli) [inline CONTRIBUTING.rst]: committing

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -74,3 +74,24 @@ Before you submit a pull request, check that it meets these guidelines:
...
+
+    $ bump2version patch
+    $ git push -u origin HEAD
+
+Open a Pull Request on GitHub and request approval from another commiter. Once your PR has been merged, you can create the release tag and push an .egg to PyPi via twine.
```
Comment (davisagli) [inline CONTRIBUTING.rst]: committer, PyPI, "push the artifacts" instead of "push an .egg" (there's a wheel too)

File: Makefile [hunk 4]
```diff
@@ -46,24 +47,24 @@ clean-test: ## remove test and coverage artifacts
...
+	rm -f output.xml
+	rm -f report.html
 
 lint: ## check style with flake8
 	flake8 cumulusci tests
 
 test: ## run tests quickly with the default Python
-	
-		python setup.py test
+	nosetests
```
Comment (davisagli) [inline Makefile]: we should switch this to run tox in the Python 3 branch.

File: CONTRIBUTING.rst (hunk 3 above)
Comment (davisagli) [reply CONTRIBUTING.rst]: Can we link to instructions for setting up your ~/.pypirc?

**Comment by cdcarter** [comment]
*finds spellcheck plugin for vscode* sheesh.

thanks.

**Pull request merged**